hkm/views/views.py

```diff
@@ -475,27 +475,61 @@ def get_template_names(self):
     def setup(self, request, *args, **kwargs):
         # Session expires after browser is closed
         request.session.set_expiry(0)
-        self.search_term = request.GET.get('search', '')
-        self.page = int(request.GET.get('page', 1))
+        self.url_params = {
+            "search": request.GET.get('search', ''),
+            "page": int(request.GET.get('page', 1)),
+            "date_from": request.GET.get('date_from', ''),
+            "date_to": request.GET.get('date_to', ''),
+            "author": filter(None, request.GET.getlist('author', None)),
+            "date": filter(None, request.GET.getlist('date', None))
+        }
         return True
 
     def handle_search(self, request, *args, **kwargs):
-        LOG.debug('Search', extra={'data': {'search_term': self.search_term, 'page': self.page}})
+        LOG.debug('Search', extra={'data': {'search_term': self.url_params['search'], 'page': self.url_params['page']}})
+
+        url_params = request.session.get('url_params', {})
```

Comment:
Instead of making the big function even bigger, there could be a couple of private functions like `__has_search_params_changes`and `__update_search_params` that takes care of comparing the current situation to the previous one and updating the params.